internal/repository/file.go

```diff
@@ -1,17 +1,21 @@
 package repository
 
 import (
-	"fmt"
 	"io/ioutil"
 	"os"
 	"path"
 	"path/filepath"
+	"strings"
 
 	"fyne.io/fyne"
 	"fyne.io/fyne/storage"
 	"fyne.io/fyne/storage/repository"
 )
 
+// fileSchemePrefix is used for concatenating with paths when we must construct
+// them from strings.
+const fileSchemePrefix string = "file://"
```

Comment:
Rather than string concatenation then passing to `ParseURI` it would be cleaner to go straight to `NewFileURI`.

Reply:
That is a good point. I think I agree that we should do that.

Reply:
Worth noting that `NewFileURI()` can silently eat errors in some cases, since the existing API did not let us return an error. By using `ParseURI()`, we have the opportunity to bubble up the error.

That said, I struggle to imagine the case where we fail to create a URI for a valid file path.

Reply:
Implemented this in c294f1620b5489bee2ed93bccef2310c42f5fc31. If you are OK with the above, you can just close this conversation.

Reply:
Yeah looks good thanks